pulse2percept/models/base.py

```diff
@@ -346,11 +346,21 @@ def predict_percept(self, implant, t_percept=None):
             # Calculate the Stimulus at requested time points:
             if t_percept is not None:
                 # Save electrode parameters
-                stim = Stimulus(stim)  # make sure stimulus is in proper format
+                if not isinstance(stim, Stimulus):
+                    stim = Stimulus(stim)  # make sure stimulus is in proper format
```

Comment:
I know `stim = Stimulus(stim)` was here before, but is it actually needed? It looks like `stim` is created from a deepcopy of `implant.stim` and there's calls to e.g. `stim.is_compressed` before, so I think we can safely remove these two lines here

Reply:
Good catch. This was added in #396 to make sure the stimulus metadata was associated with each electrode for the `BiphasicAxonMapModel`. But since then, `BiphasicAxonMapModel ` has been changed to have its own `spatial.predict_percept`, so this is obsolete now